src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileDirectoryStep.java

```diff
@@ -64,8 +64,8 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
             // Lookup the file - on a recovery, we may have already created it, but not
             // finished. Or it might already exist, created by someone else.
             FireStoreDirectoryEntry existingEntry = fileDao.lookupDirectoryEntryByPath(dataset, targetPath);
-            if (existingEntry == null) {
-                // Not there - create it
+            if (ingestFileAction.equals("createEntry")) {
```

Comment:
Replace the magic strings in this file with the constants in `ValidateIngestFileDirectoryStep`.
```suggestion
            if (ingestFileAction.equals(ValidateIngestFileDirectoryStep.createEntryAction)) {
```